pkg/cmd/pulumi/templates/cloud.go

```diff
@@ -138,6 +138,11 @@ func (s *Source) getCloudTemplates(
 }
 
 func (s *Source) getRegistryTemplates(ctx context.Context, e env.Env, templateName string) {
+	b := s.getBackend(ctx, e)
+	if !s.supportsCloudTemplates(ctx, b) {
```

Comment:
Probably worth a brief explanation that we don't support registry templates for the DIY backend, since we do support registry packages for the DIY backend.